embedding/cpp/flutter_app.cc

```diff
@@ -0,0 +1,198 @@
+}
+
+void FlutterApp::OnTerminate() {
+  assert(IsRunning());
+
+  TizenLog::Debug("Shutting down the application...");
+
+  FlutterDestroyWindow(handle);
+  handle = nullptr;
+}
+
+void FlutterApp::OnLowMemory(app_event_info_h event_info) {
+  assert(IsRunning());
+
+  FlutterNotifyLowMemoryWarning(handle);
+}
+
+void FlutterApp::OnLanguageChanged(app_event_info_h event_info) {
+  assert(IsRunning());
+
+  FlutterNotifyLocaleChange(handle);
+}
+
+void FlutterApp::OnRegionFormatChanged(app_event_info_h event_info) {
+  assert(IsRunning());
+
+  FlutterNotifyLocaleChange(handle);
+}
+
+int FlutterApp::Run(int argc, char **argv) {
+  ui_app_lifecycle_callback_s lifecycle_cb = {};
+  lifecycle_cb.create = [](void *data) -> bool {
+    FlutterApp *app = (FlutterApp *)data;
+    return app->OnCreate();
+  };
+  lifecycle_cb.resume = [](void *data) {
+    FlutterApp *app = (FlutterApp *)data;
+    app->OnResume();
+  };
+  lifecycle_cb.pause = [](void *data) {
+    FlutterApp *app = (FlutterApp *)data;
+    app->OnPause();
+  };
+  lifecycle_cb.terminate = [](void *data) {
+    FlutterApp *app = (FlutterApp *)data;
+    app->OnTerminate();
+  };
+  lifecycle_cb.app_control = [](app_control_h a, void *data) {
+    FlutterApp *app = (FlutterApp *)data;
+    app->OnAppControlReceived(a);
+  };
+
+  app_event_handler_h handler;
+  ui_app_add_event_handler(
+      &handler, APP_EVENT_LOW_MEMORY,
+      [](app_event_info_h e, void *data) {
+        FlutterApp *app = (FlutterApp *)data;
+        app->OnLowMemory(e);
+      },
+      this);
+  ui_app_add_event_handler(
+      &handler, APP_EVENT_LOW_BATTERY,
+      [](app_event_info_h e, void *data) {
+        FlutterApp *app = (FlutterApp *)data;
+        app->OnLowBattery(e);
+      },
+      this);
+  ui_app_add_event_handler(
+      &handler, APP_EVENT_LANGUAGE_CHANGED,
+      [](app_event_info_h e, void *data) {
+        FlutterApp *app = (FlutterApp *)data;
+        app->OnLanguageChanged(e);
+      },
+      this);
+  ui_app_add_event_handler(
+      &handler, APP_EVENT_REGION_FORMAT_CHANGED,
+      [](app_event_info_h e, void *data) {
+        FlutterApp *app = (FlutterApp *)data;
+        app->OnRegionFormatChanged(e);
+      },
+      this);
+  ui_app_add_event_handler(
+      &handler, APP_EVENT_DEVICE_ORIENTATION_CHANGED,
+      [](app_event_info_h e, void *data) {
+        FlutterApp *app = (FlutterApp *)data;
+        app->OnDeviceOrientationChanged(e);
+      },
+      this);
+
+  int ret = ui_app_main(argc, argv, &lifecycle_cb, this);
+  if (ret != APP_ERROR_NONE) {
+    TizenLog::Error("Could not launch an application. (%d)", ret);
+  }
+  return ret;
+}
+
+FlutterDesktopPluginRegistrarRef FlutterApp::GetRegistrarForPlugin(
+    const std::string &plugin_name) {
+  if (handle) {
+    return FlutterDesktopGetPluginRegistrar(handle, plugin_name.c_str());
+  }
+  return nullptr;
+}
+
+void FlutterApp::ParseEngineArgs() {
+  char *app_id;
+  if (app_get_id(&app_id) != 0) {
+    return;
+  }
```

Comment:
Maybe adding error log will be valuable for debugging this case? (I didn't compile below suggestion, just written it here by hand).
```suggestion
  int ret = app_get_id(&app_id);
  if (ret != APP_ERROR_NONE) {
    TizenLog::Error("Error gathering application id: %d (%s)", ret, get_error_message(ret));
    return;
  }
```

Reply:
Thanks! I'll fix.

Reply:
@swift-kim is it done?

Reply:
Done!